demo/admin/src/products/helpers/PublishAllProducts.tsx

```diff
@@ -0,0 +1,57 @@
+import { gql, useApolloClient } from "@apollo/client";
+import { CancelButton, CrudMoreActionsMenuContext } from "@comet/admin";
+import { Online } from "@comet/admin-icons";
+import { Button, Dialog, DialogActions, DialogContent, DialogTitle, ListItemIcon, MenuItem } from "@mui/material";
+import { useContext, useState } from "react";
+
+import { GQLPublishAllProductsMutation, GQLPublishAllProductsMutationVariables } from "./PublishAllProducts.generated";
+
+export function PublishAllProducts() {
+    const [open, setOpen] = useState(false);
+    const client = useApolloClient();
+    const { closeMenu } = useContext(CrudMoreActionsMenuContext);
```

Comment:
@jamesricky I need to use the context instead of the component here because the dialog would close early because the menu isn't rendered anymore. Should we export both the context and the component? The component could still be useful for most cases.

Reply:
> Should we export both the context and the component?

Yes 👍🏼